core/include/detray/grids/grid2.hpp

```diff
@@ -23,19 +23,26 @@ namespace detray {
  * @tparam serialzier_type  type of the serializer to the storage represenations
  *
  **/
-template <typename populator_type, typename axis_p0_type, typename axis_p1_type,
+template <template <template <typename...> class, template <typename...> class,
+                    template <typename, unsigned int> class, typename, auto...>
+          class populator_type,
+          typename axis_p0_type, typename axis_p1_type,
           typename serializer_type,
+          template <typename...> class vector_type = dvector,
+          template <typename...> class jagged_vector_type = djagged_vector,
           template <typename, unsigned int> class array_type = darray,
           template <typename...> class tuple_type = dtuple,
-          template <typename> class vector_type = dvector>
+          typename value_type = dindex, auto... populator_options>
 class grid2 {
 
     public:
-    using populator_t = populator_type;
+    using populator_t =
+        populator_type<vector_type, jagged_vector_type, array_type, value_type,
+                       populator_options...>;
```

Comment:
It is not for this PR to change. And in any case, this is a very subjective thing.

Since the standard library types tend to use `<foo>_type` typedefs, if we wanted to make this code resemble the `std::` classes a bit more, we should change this naming. To call the template parameters as `<foo>_t`, and the "typedefs" in the class `<foo>_type`.

But this is of course a very minor thing.

Reply:
oh... detray was doing perfectly in opposite way... It's not just the problem of this PR but entire project :(